app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,608 @@
+                const isSearch = true;
+                const {channels: receivedChannels} = await searchChannels(
+                    serverUrl, term, currentTeamId, isSearch);
+
+                if (receivedChannels) {
+                    setSearchResults(receivedChannels);
+                }
+            } else if (dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+                await searchDynamicOptions(text);
+            }
+
+            setLoading(false);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [dataSource, term]);
+
+    const filterSearchData = (source: string, searchData: DataType, searchTerm: string) => {
+        if (!data) {
+            return [];
+        }
+
+        const lowerCasedTerm = searchTerm.toLowerCase();
+
+        if (source === ViewConstants.DATA_SOURCE_USERS) {
+            return filterProfilesMatchingTerm(searchData as UserProfile[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_CHANNELS) {
+            return filterChannelsMatchingTerm(searchData as Channel[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            return searchData;
+        }
+
+        return (searchData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().startsWith(lowerCasedTerm));
+    };
+
+    const getMultiselectData = useCallback((): Selection => {
+        let myItems;
+        let multiselectItems: Selection = [];
+
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                myItems = multiselectSelected as Dictionary<UserProfile>;
+                multiselectItems = multiselectItems as UserProfile[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelected as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelected as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected, dataSource]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit, multiselectSelected]);
```

Comment:
onHandleMultiselectSubmit is not a callback, so this is updating the navigation button on every render. Also, not sure why the `multiselectSelected` is passed here as a dependency.

Reply:
When the button is pressed, It won't return the multiselect data unless I put it here, It will return empty because it's not updated after the first set.

Reply:
I guess it will be a dependency of the callback then